app/screens/channel_info/index.js

```diff
@@ -71,13 +71,15 @@ function mapStateToProps(state) {
 
     let status;
     let isBot = false;
+    let isTeammateGuest = false;
     if (currentChannel.type === General.DM_CHANNEL) {
         const teammateId = getUserIdFromChannelName(currentUserId, currentChannel.name);
         const teammate = getUser(state, teammateId);
         status = getStatusForUserId(state, teammateId);
         if (teammate && teammate.is_bot) {
             isBot = true;
         }
+        isTeammateGuest = isGuest(teammate);
         if (isGuest(teammate)) {
             currentChannelGuestCount = 1;
         }
```

Comment:
your change "works" here. but I believe this should be re-written to

```js
if (teammate) {
    isBot = teammate.is_bot || falsel
    if (isGuest(teammate)) {
       isTeammateGuest = true;
       currentChannelGuestCount = 1;
    }
}

Reply:
:man_facepalming: